packages/relay/src/lib/eth.ts

```diff
@@ -624,15 +657,31 @@ export class EthImpl implements Eth {
     const latestBlockTolerance = 1;
     this.logger.trace(`${requestIdPrefix} getBalance(account=${account}, blockNumberOrTag=${blockNumberOrTag})`);
 
+    let latestBlock: LatestBlockNumberTimestamp | null | undefined;
     // this check is required, because some tools like Metamask pass for parameter latest block, with a number (ex 0x30ea)
     // tolerance is needed, because there is a small delay between requesting latest block from blockNumber and passing it here
     if (!EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
-      const latestBlock = await this.blockNumber(requestId);
-      const blockDiff = Number(latestBlock) - Number(blockNumberOrTag);
+      const cacheKey = `${constants.CACHE_KEY.ETH_BLOCK_NUMBER}`;
+      const blockNumberCached = this.cache.get(cacheKey);
+      
+      if(blockNumberCached) {
+        this.logger.trace(`${requestIdPrefix} returning cached value ${cacheKey}:${JSON.stringify(blockNumberCached)}`);
+        latestBlock = { blockNumber: blockNumberCached, timeStampTo: '0' };
+
+      } else {
+        latestBlock = await this.blockNumberTimestamp(requestId);
```

Comment:
do we not want to add to the cache here?  Maybe, I'm misunderstanding what is happening.

Reply:
We could but it would require its own separate cache key, and value.  The old getBlock method also sets the cache and is called in many places, so by the time metamask connects and calls getBalance for the first time, the cache has a value in it.  Using a unique cache for getBalance would mean that it gets populated on the first call to getBalance only which I am afraid would cause the behavior I was seeing before this fix.  Metamask passed in the latest block using the block number and not the tag, so the small delay in look up caused an inconsistent result.  It defeated the purpose of the blocktolerance setting.